### DISCUSSION TITLE
Prioritise arguments, events based on latest user interaction

### DISCUSSION THREAD

**Discussion by Nayananga** [h2oai/wave#1094] [Ideas]
Hi, I've been implementing wave apps for a while. It is excellent and looks nice. But when I have multiple components and handlers associated with them, and when the user does a particular interaction, sometimes the incorrect handler gets triggered. 

![Screenshot 2021-10-28 at 11 57 22](https://user-images.githubusercontent.com/25351304/139200541-d6c729d7-04db-4947-b7d3-1e0d930a55fa.png)

What I understood is this is because wave executes q.args and q.events in the order it has saved. For example, if there are two q.args and when the WAVE wants to choose what needs to handle first, it is best to select the most recent argument.

This can be solved by maintaining arguments in a stack, associating timestamps with arguments or adding weights according to the user's latest interaction. More ideas are welcome.

**Comment by sairaamv** [comment]
What do you mean by incorrect handler, getting triggered? 
Just curios where incorrect handlers are triggered. Can you showcase an example? -- thanks

**Reply by Nayananga** [reply]
@sairaamv @lo5 In the screenshot above, you may see that `select_mode_dropdown` and `project_tabs` are arguments that carry values at the moment. There was a dropdown menu in which I had selected some value, and later, I wanted to switch to a different tab view. But in WAVE, it triggers the dropdown handler instead of the `project_tabs` handler. So I have to put an if-else statement inside the dropdown handler and switch into the `project_tabs` handler which I actually want to handle. This is okay for one or two cases, but as you can see, there are multiple dropdown items and textboxes in this q.args that are not activated at this moment. Still, I have to handle all those arguments similarly, so from the developer perspective, it is irritating and still does not feel right.
